ReactSkia/components/RSkComponentImage.cpp

```diff
@@ -181,6 +226,33 @@ void RSkComponentImage::requestNetworkImageData(ImageSource source) {
 
   folly::dynamic query = folly::dynamic::object();
 
+  //Before network request, reset the cache info with default values
+  canCacheData_ = true;
+  cacheExpiryTime_ = DEFAULT_MAX_CACHE_EXPIRY_TIME;
+
+  // headercallback lambda fuction
+  auto headerCallback =  [&](void* curlresponseData,void *userdata)->bool {
+    CurlResponse *responseData =  (CurlResponse *)curlresponseData;
+    CurlRequest *curlRequest = (CurlRequest *) userdata;
+
+    double responseMaxAgeTime = DEFAULT_MAX_CACHE_EXPIRY_TIME;
+    double requestMaxAgeTime = DEFAULT_MAX_CACHE_EXPIRY_TIME;
+    // Parse server response headers and retrieve caching details
+    auto responseCacheControlData = responseData->headerBuffer.find("Cache-Control");
+    if(responseCacheControlData != responseData->headerBuffer.items().end()) {
+      std::string responseCacheControlString = responseCacheControlData->second.asString();
+      canCacheData_ = shouldCacheData(responseCacheControlString);
+      if(canCacheData_) responseMaxAgeTime = getCacheMaxAgeDuration(responseCacheControlString);
+    }
+
+    // TODO : Parse request headers and retrieve caching details
+
+    cacheExpiryTime_ = std::min(std::min(responseMaxAgeTime,requestMaxAgeTime),static_cast<double>(DEFAULT_MAX_CACHE_EXPIRY_TIME));
```

Comment:
since we are not handling request headers now , can we avoid the calculation of requestMaxAgeTime here ?